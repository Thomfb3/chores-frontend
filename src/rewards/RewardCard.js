import React, { useContext } from "react";
import { Link } from "react-router-dom";
import { formatDate } from "../helpers/formatDate";
import findUserInTeam from "../helpers/findUserInTeam";
import UserContext from "../auth/UserContext";

function RewardCard({
    id,
    title,
    description,
    points,
    status,
    sponsor,
    createdBy,
    createdAt
}) {

    console.debug("RewardCard");
    const { currentTeamUsers } = useContext(UserContext);
    const createdByUser = findUserInTeam(createdBy, currentTeamUsers);
    const foundSponsor = findUserInTeam(sponsor, currentTeamUsers);

    return (
        <Link to={`/rewards/${id}`} className="RewardCard">
            <div className="RewardCard__container">
                <div className="RewardCard__inner-container RewardCard__inner-container--align-center">
                    <div className="RewardCard__title-grow">
                        <p className="RewardCard__title">{title}</p>
                        <p className="RewardCard_description">{description}</p>
                    </div>
                    <div className="RewardCard__points-section">
                        <p>Points Required</p>
                        <p className="RewardCard__points">{points} pts</p>
                    </div>
                    <div className="RewardCard__sponsor-section">
                        <p className="RewardCard__status">{status === "open" ? "Available" : "Currently Unavailable"}</p>
                        <p className="RewardCard__sponsor-label">Sponsored by:</p>
                        <p className="RewardCard__sponsor">{foundSponsor.username}</p>
                    </div>
                </div>
                <p className="RewardCard__footer">{`Reward was created at ${formatDate(createdAt)}, by ${createdByUser.username}`}</p>
            </div>
        </Link>
    );
};

export default RewardCard;




