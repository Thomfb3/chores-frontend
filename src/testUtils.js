import React from "react";
import UserContext from "./auth/UserContext";
import ChoreCommentContext from "./comments/ChoreCommentContext";

const demoUser = {
  _id: "userid1",
  username: "testuser1",
  firstName: "testfirst1",
  lastName: "testlast1",
  profileImage: "defaultProfile.jpg",
  allTimePoints: 1000,
  currentPoints: 200,
  role: "admin"
};

const testTeam = {
  _id: "testTeam",
  name: "testTeam"
};

const testTeamUsers = [
  {
    _id: "userid1",
    username: "testuser1",
    firstName: "testfirst1",
    lastName: "testlast1",
    profileImage: "defaultProfile.jpg",
    allTimePoints: 1000,
    currentPoints: 200,
    role: "admin"
  },
  {
    _id: "userid2",
    username: "testuser2",
    firstName: "testfirst2",
    lastName: "testlast2",
    profileImage: "defaultProfile.jpg",
    allTimePoints: 1000,
    currentPoints: 200,
    role: "user"
  },
  {
    _id: "userid3",
    username: "testuser3",
    firstName: "testfirst3",
    lastName: "testlast3",
    profileImage: "defaultProfile.jpg",
    allTimePoints: 1000,
    currentPoints: 200,
    role: "user"
  },
  {
    _id: "userid4",
    username: "testuser4",
    firstName: "testfirst4",
    lastName: "testlast4",
    profileImage: "defaultProfile.jpg",
    allTimePoints: 1000,
    currentPoints: 200,
    role: "user"
  }
];

let choreComments = [];

const UserProvider =
  ({ children,
    currentUser = demoUser,
    currentTeam = testTeam,
    currentTeamUsers = testTeamUsers,
    isAdmin = false
  }) => (
      <UserContext.Provider value={{ currentUser, currentTeam, currentTeamUsers }}>
        {children}
      </UserContext.Provider>
    );


const CommentProvider =
  ({ children,
    currentChoreComments = choreComments
  }) => (
      <ChoreCommentContext.Provider value={{ currentChoreComments }}>
        {children}
      </ChoreCommentContext.Provider>
    );

export { UserProvider, CommentProvider };

