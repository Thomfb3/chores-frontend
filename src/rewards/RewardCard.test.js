import React from "react";
import { render } from "@testing-library/react";
import RewardCard from "./RewardCard";
import { UserProvider } from "../testUtils";
import { MemoryRouter } from "react-router";


it("matches snapshot", function () {
  
    const { asFragment } = render(
        <MemoryRouter>
            <UserProvider>
                <RewardCard
                    id="rewardid1"
                    key="rewardid1"
                    title="Fried Chicken"
                    description="bucket of fried chicken"
                    points={1000}
                    status="available"
                    sponser="userid1"
                    createdBy="userid1"
                    createdAt="2021-11-05T03:47:24.581+00:00"
                />
            </UserProvider>
        </MemoryRouter>,
    );
    expect(asFragment()).toMatchSnapshot();
});
