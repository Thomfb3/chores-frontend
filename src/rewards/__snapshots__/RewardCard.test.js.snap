// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <a
    class="RewardCard"
    href="/rewards/rewardid1"
  >
    <div
      class="RewardCard__container"
    >
      <div
        class="RewardCard__inner-container RewardCard__inner-container--align-center"
      >
        <div
          class="RewardCard__title-grow"
        >
          <p
            class="RewardCard__title"
          >
            Fried Chicken
          </p>
          <p
            class="RewardCard__description"
          >
            bucket of fried chicken
          </p>
        </div>
        <div
          class="RewardCard__points-section"
        >
          <p>
            Points Required
          </p>
          <p
            class="RewardCard__points"
          >
            1000 pts
          </p>
        </div>
        <div
          class="RewardCard__sponsor-section"
        >
          <p
            class="RewardCard__status"
          >
            Currently Unavailable
          </p>
          <p
            class="RewardCard__sponsor-label"
          >
            Sponsored by:
          </p>
          <p
            class="RewardCard__sponsor"
          />
        </div>
      </div>
      <p
        class="RewardCard__footer"
      >
        Reward was created at Thursday, November 4, 2021 - 11:47 PM, by testuser1
      </p>
    </div>
  </a>
</DocumentFragment>
`;
